tests/test.spec.ts

```diff
@@ -0,0 +1,294 @@
+  if (!atLine && !removeLines) {
+    fs.appendFileSync(filePath, content.text + '\n\n');
+  } else {
+    const lines = fs.readFileSync(filePath, 'utf8').split('\n');
+    if (removeLines) {
+      removeLines.forEach((lineNumber) => {
+        lines[lineNumber - 1] = '~~~REMOVE~~~';
+      });
+    }
+    if (atLine) {
+      lines.splice(atLine - 1, 0, content.text);
+    }
+    const modifiedContent = lines
+      .filter((line) => line !== '~~~REMOVE~~~')
+      .join('\n');
+
+    fs.writeFileSync(filePath, modifiedContent, 'utf8');
+  }
+}
+
+async function compareToFile(page: Page, buttonName: string, pathName: string) {
+  const expected = await clickCopyButton(page, buttonName);
+  const actual = fs.readFileSync(pathName, { encoding: 'utf8' });
+  compareOutputs(expected.text, actual);
+}
+
+async function setupFolders(projectFolder: string) {
+  if (!fs.existsSync('guides-testing')) {
+    await fs.promises.mkdir('guides-testing');
+  }
+  const projectPath = `guides-testing/${projectFolder}`;
+  if (fs.existsSync(projectPath)) {
+    await fs.promises.rm(projectPath, {
+      recursive: true,
+      force: true,
+    });
+  }
+}
+
+function separateCommand(text: string): Instruction {
+  const lines = text.trim().split('\n');
+  // The first line contains the command
+  const command = lines[0].replace('$', '').trim();
+  // The rest of the lines contain the console output
+  const consoleOutput = lines.slice(1).join('\n');
+  return {
+    text: command,
+    output: consoleOutput,
+  };
+}
+
+function compareOutputs(expected: string, actual: string) {
+  const split1 = expected
+    .trim()
+    .split(EOL)
+    .filter((line) => !line.startsWith('// ANCHOR'));
+  const split2 = actual.trim().split(EOL);
+  expect(split1.length === split2.length).toBeTruthy();
+  split1.forEach((line, i) => {
+    const trimmedLineA = line.trim().replace(/\u00A0/g, ' ');
+    const trimmedLineB = split2[i].trim().replace(/\u00A0/g, ' ');
+    expect(trimmedLineA).toEqual(trimmedLineB);
+  });
+}
+
+async function compareFiles(testPathName: string, refPathName: string) {
+  const actual = fs.readFileSync(testPathName, { encoding: 'utf8' });
+  const expected = fs.readFileSync(refPathName, { encoding: 'utf8' });
+  compareOutputs(expected, actual);
+}
+
+function checkIfIsIncremented(initialIndex: number, finalIndex: number) {
+  const initial: number = parseInt(saved[initialIndex]);
+  const final: number = parseInt(saved[finalIndex]);
+  console.log('INITIAL:', initial);
+  console.log('FINAL:', final);
+  const isIncremented = final === initial + 1;
+  expect(isIncremented).toBeTruthy();
+}
+
+function stopServers() {
+  console.log('STOPPING SERVERS');
+  // stop & delete pm2 servers
+  execSync(STOP_SERVERS, {
+    encoding: 'utf-8',
+  });
+  console.log('DONE STOPPING SERVERS');
+}
+
+function checkIfServersRunning() {
+  try {
+    const output = execSync('pm2 list --no-color').toString();
```

Comment:
Why we use pm2 and not node directly? If the need is to put on background and kill after you can use spawn.

Reply:
I tried with node/spawn and it was just kind of messy, I think with pm2 it is simpler and cleaner